test-suite/exonum-py-tests/exonum_tests/migrate.py

```diff
@@ -0,0 +1,280 @@
+import time
+import unittest
+
+from exonum_client import ExonumClient
+from exonum_launcher.action_result import ActionResult
+from exonum_launcher.configuration import Configuration
+from exonum_launcher.explorer import NotCommittedError
+from exonum_launcher.launcher import Launcher
+
+from suite import (
+    assert_processes_exited_successfully,
+    run_4_nodes,
+    wait_network_to_start,
+    generate_config,
+    generate_migration_config,
+)
+
+INSTANCE_NAME = "cryptocurrency"
+
+
+class MigrationTests(unittest.TestCase):
+    """Tests for a checking service migration mechanism."""
+
+    def setUp(self):
+        self.network = run_4_nodes("cryptocurrency-migration")
+        self.addCleanup(self._tear_down, False)
+        wait_network_to_start(self.network)
+
+    def wait_for_api_restart(self):
+        """Waits until the API servers of nodes are restarted after the set
+        of active services has changed."""
+
+        time.sleep(0.25)
+        wait_network_to_start(self.network)
+
+    def test_full_migration_flow(self):
```

Comment:
Would it be too difficult to generate some service data before migration and check that it persists after migration? (Maybe, reuse the code from `crypto_advanced` test?)

Reply:
Good point. Definitely makes sense 👍🏻